aws-logs-loggroup/aws-logs-loggroup.json

```diff
@@ -3,8 +3,14 @@
     "description": "Resource schema for AWS::Logs::LogGroup",
     "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs.git",
     "properties": {
+        "KmsKeyId": {
```

Comment:
Will need to update the `handlers` section too with additional IAM keys here based on the API decision.

Reply:
I'm assuming you will need "logs:AssociateKmsKey" if you do it in one action too. It's not yet in the docs (https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/permissions-reference-cwl.html)

Reply:
On that note. Why is `"logs:PutRetentionPolicy"` a permission for the UpdateHandler, but not for the CreateHandler? They both do that in a separate call ([see here](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/c72bc6d300a1ba3db06862eb22e7f81731fc9db0/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/CreateHandler.java#L50)):

Reply:
Unclear what permissions are needed here. Sometimes the explicit API permission isn't checked when wrapped in another API call.

On the second point, that's totally a miss. This type wouldn't work when run in Private mode.

Reply:
You were right, `logs:CreateLogGroup` is all you need